src/vm/builtins/Array.zig

```diff
@@ -60,9 +61,34 @@ fn array_reverse(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value
     return Code.Value.Object.Array.init(vm.allocator, new_array) catch Code.Value{ .none = {} };
 }
 
-fn length(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value {
-    _ = vm;
+fn foreach(vm: *VirtualMachine, arguments: []const Code.Value) Code.Value {
+    if (!((arguments[0] == .object and (arguments[0].object == .array or arguments[0].object == .map)) and (arguments[1] == .object and arguments[1].object == .function))) {
+        return Code.Value{ .none = {} };
+    }
+
+    const callback = arguments[1].object.function;
+
+    if (arguments[0].object == .map) {
+        const map = arguments[0].object.map;
+        var map_iter = map.inner.iterator();
+
+        while (map_iter.next()) |map_entry| {
```

Comment:
map_entry

Reply:
what's wrong?

Reply:
my suggestion is to use `map_iter` for the iterator and `entry` as variable names instead